### PR TITLE
Add definition provider sample plugin

### PR DESCRIPTION
**Pull request by mmorhun**
Demo:
go-to-definition

### CODE REVIEW

File: definition-provider-plugin/src/definition-provider-plugin-frontend-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Sample plugin which demonstrates definition provider plugin API.
+ *
+ * Searches for function stop in current document and if it exists adds the function as a definition source (for any symbol).
+ */
+
+import * as theia from '@theia/plugin';
+
+const disposables: theia.Disposable[] = [];
```
Comment (benoitf) [inline definition-provider-plugin/src/definition-provider-plugin-frontend-plugin.ts]: same comment here than before (the start context should be used)

File: definition-provider-plugin/src/definition-provider-plugin-frontend-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if (index !== -1) {
+            return new theia.Range(i, index + functionPrefix.length,
+                i, index + functionPrefix.length + name.length);
+        }
+    }
+
+    return undefined;
+}
+
+export function stop() {
```
Comment (benoitf) [inline definition-provider-plugin/src/definition-provider-plugin-frontend-plugin.ts]: to be empty

File: definition-provider-plugin/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        "watch": "webpack-cli -w --config webpack.config.js",
+        "compile": "webpack-cli --config webpack.config.js",
+
+        "build": "yarn run format-code && yarn run compile && theia:plugin pack"
+      },
+      "engines": {
+        "theiaPlugin": "next"
+      },
+      "theiaPlugin": {
+        "frontend": "dist/definition-provider-plugin-frontend-plugin.js"
```
Comment (benoitf) [inline definition-provider-plugin/package.json]: why frontend ?

File: definition-provider-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+      "name": "definition-provider-plugin-plugin",
+      "publisher": "theia",
+      "keywords": [
+        "theia-plugin"
+      ],
+      "version": "0.0.1",
+      "license": "[object Object]",
```
Comment (benoitf) [inline definition-provider-plugin/package.json]: I think there is an issue there on the license

Comment (mmorhun) [reply definition-provider-plugin/package.json]: Seems so...

File: definition-provider-plugin/package.json (hunk 3 above)
Comment (mmorhun) [reply definition-provider-plugin/package.json]: Easier to debug

Comment (mmorhun) [reply definition-provider-plugin/package.json]: I created this plugin and some others for testing of new Theia plugin API methods.

Comment (mmorhun) [reply definition-provider-plugin/package.json]: Just left one provider to demonstrate abilities of the API.

File: definition-provider-plugin/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        "webpack": "^4.1.1",
+        "webpack-cli": "^3.1.1"
+      },
+      "scripts": {
+        "prepare": "yarn run clean && yarn run build",
+        "clean": "rimraf lib",
+        "format-code": "tsfmt -r",
+        "watch": "webpack-cli -w --config webpack.config.js",
+        "compile": "webpack-cli --config webpack.config.js",
+
```
Comment (benoitf) [inline definition-provider-plugin/package.json]: extra line

**Pull request merged**